svm-message-processor/src/lib.rs

```diff
@@ -0,0 +1,680 @@
+use {
+    solana_measure::measure_us,
+    solana_program_runtime::invoke_context::InvokeContext,
+    solana_sdk::{
+        account::WritableAccount,
+        precompiles::get_precompile,
+        saturating_add_assign,
+        sysvar::instructions,
+        transaction::TransactionError,
+        transaction_context::{IndexOfAccount, InstructionAccount},
+    },
+    solana_svm_transaction::svm_message::SVMMessage,
+    solana_timings::{ExecuteDetailsTimings, ExecuteTimings},
+};
+
+/// Process a message.
+/// This method calls each instruction in the message over the set of loaded accounts.
+/// For each instruction it calls the program entrypoint method and verifies that the result of
+/// the call does not violate the bank's accounting rules.
+/// The accounts are committed back to the bank only if every instruction succeeds.
```

Comment:
Let's leave any comments about comitting and banks out of this, they are a higher-level concept that really isn't relevant here (imo).

What is important here is that a `TransactionError` is returned if **any** of the instructions fail

Reply:
Oh I didn't write this I just copied the associated function along with its doc comment from here: https://github.com/anza-xyz/agave/blob/990f938a718d4bcc791e26002a5475a233afa4c8/svm/src/message_processor.rs#L30

Want me to change it while we're here?

Reply:
yes please 😄